device/sycl/include/traccc/sycl/clusterization/component_connection.hpp

```diff
@@ -0,0 +1,49 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// SYCL library include(s).
+#include "traccc/sycl/utils/queue_wrapper.hpp"
+
+// Project include(s).
+#include "traccc/clusterization/detail/sparse_ccl.hpp"
+#include "traccc/edm/cell.hpp"
+#include "traccc/edm/cluster.hpp"
+#include "traccc/utils/algorithm.hpp"
+
+namespace traccc::sycl {
+
+struct component_connection
+    : public algorithm<host_cluster_container(const host_cell_container &)> {
+
+    public:
+    /// Constructor for component_connection
+    ///
+    /// @param mr is memory resource
+    /// @param queue is the sycl queue for kernel inocation
+    component_connection(vecmem::memory_resource &mr, queue_wrapper queue);
+
+    /// Callable operator for component connection for all the cells per event
+    ///
+    /// @param cells_per_event are the input cells grouped by module
+    host_cluster_container operator()(
+        const host_cell_container &c) const override;
+
+    /// Callable operator for component connection for all the cells per event
+    ///
+    /// @param cells_per_event are the input cells grouped by module
+    /// @return a container of clusters from all the modules
+    void operator()(const host_cell_container &cells_per_event,
+                    output_type &clusters) const;
```

Comment:
It's fine to implement a function with this interface as well. But in general, this was not needed. It would be enough to implement the "standard" algorithm function.

The reason that the host version of the algorithm outsourced the execution into a function like this was to make it possible to use the same code with both host- and device containers. That is not a requirement here.

And if/when we'll start to re-think how containers are passed around, we'll need to touch this code anyway. So we might as well keep it as simple as possible for now.

Reply:
I see. I removed this extra `operator()` step.